### ISSUE TITLE
Mark Result::is_ok as #[must_use]

### ISSUE THREAD

**Issue by alex** [rust-lang/rust#59610]
I just ran into a project where `Result::is_ok` was accidentally being called without using the result -- I believe it had been confused with an assertion method like `expect()`. Had `is_ok` been marked `#[must_use]` this mistake would have been prevented.

**Comment by shepmaster** [comment]
Duplicate of #48926

**Comment by varkor** [comment]
@alex: feel free to open a pull request adding `#[must_use]` to `is_ok`.

**Comment by shepmaster** [comment]
@varkor doesn't that rather ignore the intent of the linked issue where the libs team is attempting to decide by what rationale functions should and should not have the annotation? Arbitrarily picking functions seems like a poor avenue to go down. For example, why not also `is_err`? What makes those two methods special compared to `Result::ok`? etc.

**Comment by varkor** [comment]
In the past, pull requests for individual methods have been accepted (e.g. https://github.com/rust-lang/rust/pull/50177, https://github.com/rust-lang/rust/pull/58145, https://github.com/rust-lang/rust/pull/49533). The purpose of the tracking issue seems to me to decide on an overarching set of rules for `#[must_use]`, but it seems clear that if a method is effectless and can be confused with an effectful variant, marking as `#[must_use]` is definitely what we want to do and waiting on a general guideline now just leaves the behaviour subpar (especially as the tracking issue has not seen much progress recently). Specifically in this situation, we have a concrete instance of a particular method being misused and causing practical problems.

I do think if a pull request was opened, it should include `is_err`. I agree broader questions should be left to the tracking issue. Maybe the best option would be to open a pull request and r? a libs team member, if there's any controversy.

**Comment by shepmaster** [comment]
> it seems clear that if a method is effectless and can be confused with an effectful variant, marking as `#[must_use]` is definitely what we want to do

It's probably worth attempting to say something to that effect on the tracking issue.

I do agree that it's a shame that it's moving so slow; you can even see some comments from me with  issues I've seen in the wild.
